Create direct network upload dialog

AlmightyYakob opened the pull request:
This separates uploading a table via `csv` file, and uploading a network with any of the other file types. Under the hood this abstracts the original file uploader dialog into its own component, which both the network and table uploader use. This also updates the types in `multinetjs`, and removes redundant code. Making this a draft PR so that we can iterate with @jtomeck on it.

Closes #279 

TODO:
- [ ] create a multinetjs release
- [ ] update client to use new multinetjs version

AlmightyYakob commented:
Also @waxlamp this PR depends on a `multinetjs` update to pass CI.

waxlamp commented on client/src/components/TableDialog.vue:
```diff
@@ -23,146 +23,42 @@
-    },
-  },
-
+  computed: {},
```
```suggestion
```

waxlamp commented on multinetjs/src/index.ts:
```diff
@@ -33,7 +33,13 @@ export interface EdgesSpec {
+
+export function validUploadType(ext: string): ext is UploadType {
+  return ['csv', 'nested_json', 'newick', 'd3_json'].includes(ext);
+}
```
Type narrowing!! :heart: :tada:

waxlamp commented on multinetjs/src/index.ts:
```diff
@@ -50,8 +56,8 @@ export type EdgesOptionsSpec = OffsetLimitSpec & {
-export interface UploadTableOptionsSpec {
-  type: UploadType;
+export interface FileUploadOptionsSpec {
+  type: TableUploadType | GraphUploadType;
```
This can just be

```suggestion
  type: UploadType;
```

right?

AlmightyYakob replied:
Yep

waxlamp commented:
Shouldn't this PR be rebased on #297? Otherwise it makes it difficult to test the functionality of this branch.

JackWilb commented:
I'm getting some errors trying to upload the miserables.json file. It looks like there's a problem with the attribute getter since the d3 json uploaders uses "source" and "target", not "_from" and "_to". Here's the full stack trace:
```
Traceback (most recent call last):
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 70, in sentry_patched_wsgi_app
    environ, start_response
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 126, in __call__
    reraise(*_capture_exception(hub))
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/sentry_sdk/_compat.py", line 57, in reraise
    raise value
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 122, in __call__
    _sentry_start_response, start_response, span
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 69, in <lambda>
    return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/jwilburn/.local/share/virtualenvs/multinet-xhKnJcBN/lib/python3.7/site-packages/flasgger/utils.py", line 255, in wrapper
    return function(*args, **kwargs)
  File "/home/jwilburn/Projects/multinet/multinet/uploaders/d3_json.py", line 122, in upload
    properties = util.get_edge_table_properties(workspace, edge_table_name)
  File "/home/jwilburn/Projects/multinet/multinet/util.py", line 47, in get_edge_table_properties
    from_node, to_node = edge["_from"].split("/"), edge["_to"].split("/")
KeyError: '_from'
```

AlmightyYakob commented:
@waxlamp I think I've addressed your concerns, aside from the "fat" feeling of the form. I tried making the suggested changes their, but it made the UI look a bit weird, maybe @jtomeck cant take a look. Lmk what you think of the new changes.

waxlamp commented:
This looks great, @AlmightyYakob. I opened #302 to address the "fat fields" issue.

waxlamp commented:
BTW I discovered another reason not to target PRs against other PRs: CI won't run!

AlmightyYakob commented:
> BTW I discovered another reason not to target PRs against other PRs: CI won't run!

Oh interesting, that's definitely something to remember.
